cli/azd/pkg/azdo/project.go

```diff
@@ -195,11 +195,15 @@ func GetProjectFromExisting(
 	}
 
 	projects := getProjectsResponse.Value
-	projectsList := make([]core.TeamProjectReference, len(projects))
-	options := make([]string, len(projects))
-	for idx, project := range projects {
-		options[idx] = *project.Name
-		projectsList[idx] = project
+	projectsList := make([]core.TeamProjectReference, 0, len(projects))
+	options := make([]string, 0, len(projects))
+	for _, project := range projects {
+		options = append(options, *project.Name)
+		projectsList = append(projectsList, project)
+	}
+
+	if len(options) == 0 {
+		return "", "", fmt.Errorf("no Azure DevOps projects found")
 	}
```

Comment:
Please update PR description that this go beyond just subscription lists, but other lists like this as well.

Reply:
The PR description already reflects this:

"This fix improves the error message and provides an error hint, and also improves other similar occurrences of console.Select without length checking on options."